OrbitCore/TcpClient.cpp

```diff
@@ -72,7 +72,10 @@ void TcpClient::Connect(const std::string& a_Host) {
 }
 
 void TcpClient::Stop() {
-  if (workerThread_.joinable()) {
+  const bool inWorkerThread = std::this_thread::get_id()
+          == workerThread_.get_id();
```

Comment:
interesting - I thought joinable tests this and returns false.

Reply:
Yes. That's exactly what I thought as well.

cppreference.com says
> Checks if the std::thread object identifies an active thread of execution. Specifically, returns true if get_id() != std::thread::id(). So a default constructed thread is not joinable.

There is already an error because `std::thread::id` is a type. I assumed they meant `std::this_thread::get_id()`. But maybe they actually meant:
> Specifically, returns true if get_id() *==* std::this_thread::get_id().

I have to check the standard on this.

Reply:
Ok, got it.
* `std::thread::id` is a class.
* `std::thread::id()` constructs a object of this class by calling the default constructor.
* This default-constructed id-object represents an invalid thread id.
* The`operator==` for `std::thread::id` is defined such that it returns if (not iff) the objects represent the same thread, but it also return true if both objects represent an invalid thread id.
* So `get_id() != std::thread::id{}` is something like `get_id().is_valid()` but it's not `get_id() != std::this_thread::get_id()`

Reply:
In which case, do you want to check if workerThread_.get_id() is valid? or operator== does it for you?

Reply:
I check for validity with `workerThread_.joinable()`. That's why the condition is `inWorkerThread && workerThread_.joinable()`.